mirgecom/initializers.py

```diff
@@ -0,0 +1,218 @@
+    background flow velocity (velocity).
+    """
+
+    def __init__(
+        self, beta=5, center=[0, 0], velocity=[0, 0],
+    ):
+        """Initialize vortex parameters
+
+        Parameters
+        ----------
+        beta : float
+        vortex amplitude
+        center : 2-dimensional float array
+        center of vortex
+        velocity : 2-dimensional float array
+        fixed flow velocity used for exact solution at t != 0
+        """
+
+        self._beta = beta
+        self._center = np.array(center)
+        self._velocity = np.array(velocity)
+
+    def __call__(self, t, x_vec, eos=IdealSingleGas()):
+        vortex_loc = self._center + t * self._velocity
+
+        # coordinates relative to vortex center
+        x_rel = x_vec[0] - vortex_loc[0]
+        y_rel = x_vec[1] - vortex_loc[1]
+
+        gamma = eos.gamma()
+        r = clmath.sqrt(x_rel ** 2 + y_rel ** 2)
+        expterm = self._beta * clmath.exp(1 - r ** 2)
+        u = self._velocity[0] - expterm * y_rel / (2 * np.pi)
+        v = self._velocity[1] + expterm * x_rel / (2 * np.pi)
+        mass = (1 - (gamma - 1) / (16 * gamma * np.pi ** 2) * expterm ** 2) ** (
+            1 / (gamma - 1)
+        )
+        p = mass ** gamma
+
+        e = p / (gamma - 1) + mass / 2 * (u ** 2 + v ** 2)
+
+        return flat_obj_array(mass, e, mass * u, mass * v)
+
+
+class Lump:
+    r"""Implements an N-dimensional Gaussian lump of mass.
+
+    The Gaussian lump is defined by:
+
+    .. :math::
+
+    {\rho}(r) = {\rho}_{0} + {\rho}_{a}\exp^{(1-r^{2})}\\
+    {\rho}\vec{V} = {\rho}(r)\vec{V_0}\\
+    {\rho}E = (\frac{p_0}{(\gamma - 1)} + \frac{1}{2}\rho{|V_0|}^2
+
+    Where :math:`V_0` is the fixed velocity specified
+    by the user at init time, and :math:`\gamma` is taken
+    from the equation-of-state object (eos).
+
+    A call to this object after creation/init creates
+    the lump solution at a given time (t)
+    relative to the configured origin (center) and
+    background flow velocity (velocity).
+
+    This object also functions as a boundary condition
+    by providing the "get_boundary_flux" method to
+    prescribe exact field values on the given boundary.
+
+    This object also supplies the exact expected RHS
+    terms from the analytic expression in the
+    "expected_rhs" method.
+    """
+
+    def __init__(
+        self, numdim=1, rho0=1.0, rhoamp=1.0, p0=1.0, center=[0], velocity=[0],
+    ):
+        r"""Initialize Lump parameters
+
+        Parameters
+        ----------
+        numdim : integer
+        specify the number of dimensions for the lump
+        rho0 : float
+        specifies the value of :math:`\rho_0`
+        rhoamp : float
+        specifies the value of :math:`\rho_a`
+        p0 : float
+        specifies the value of :math:`p_0`
+        center : 2-dimensional float array
+        center of lump
+        velocity : 2-dimensional float array
+        fixed flow velocity used for exact solution at t != 0
```

Comment:
I think the numpy docstring convention calls for the descriptions to be indented. (This applies here and everywhere else.)

https://numpydoc.readthedocs.io/en/latest/format.html

Reply:
Not sure I have all the docs all done properly.  I'll be fixing these as encountered.